pkg/controller/replication.go

```diff
@@ -540,22 +551,24 @@ func (s *Service) DeleteLocalVolume(ctx context.Context,
 		return nil, status.Errorf(codes.Internal, "Error: Unable to get volume for deletion")
 	}
 
-	vgs, err := arr.GetClient().GetVolumeGroupsByVolumeID(ctx, volumeID)
-	if err != nil {
-		if apiError, ok := err.(gopowerstore.APIError); !ok || !apiError.NotFound() {
-			return nil, err
+	if protocol != "nfs" {
```

Comment:
Volume API calls need to be called only for Block protocol. For NFS, there is FS API.
_arr.GetClient().GetVolume(ctx, volumeID)
arr.GetClient().DeleteVolume(ctx, nil, volumeID)_